drivers/ng_at86rf2xx/ng_at86rf2xx_netdev.c

```diff
@@ -0,0 +1,596 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ng_at86rf2xx
+ * @{
+ *
+ * @file
+ * @brief       Netdev adaption for the AT86RF2xx drivers
+ *
+ * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "net/ng_ieee802154.h"
+#include "net/ng_netbase.h"
+#include "ng_at86rf2xx.h"
+#include "ng_at86rf2xx_netdev.h"
+#include "ng_at86rf2xx_internal.h"
+#include "ng_at86rf2xx_registers.h"
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+
+/* TODO: generalize and move to ng_ieee802154 */
+static size_t _make_data_frame_hdr(ng_at86rf2xx_t *dev, uint8_t *buf,
+                                   ng_netif_hdr_t *hdr)
+{
+    int pos = 0;
+
+    /* we are building a data frame here */
+    buf[0] = NG_IEEE802154_FCF_TYPE_DATA;
+    /* default to wants_ack TODO: provide means to specify */
+    buf[0] |= NG_IEEE802154_FCF_ACK_REQ;
+    /* default to compress pan TODO: provide means to specify src PAN*/
+    buf[0] |= NG_IEEE802154_FCF_PAN_COMP;
+    if (hdr->dst_l2addr_len == 2) {
+        /* default to 2 byte addresses for src and dst */
```

Comment:
The idea was to take `NETCONF_OPT_SRC_LEN` for the length of the source address.